api/src/controllers/record-utils.js

```diff
@@ -240,17 +219,18 @@ const createRecordByJSON = data => {
     throw new PublicError('Form not found: ' + options.form);
   }
 
-  const formData = {};
-
-  // For now only save string and number fields, ignore the others.
-  // Lowercase all property names.
-  for (var k in data) {
-    if (['string', 'number'].indexOf(typeof data[k]) >= 0) {
```

Comment:
this doesn't make sense because according to documentation we support more field types than this but we just silently discard them